### DISCUSSION TITLE
How to generate a URL to a PR file (i.e. #diff-xyz)

### DISCUSSION THREAD

**Discussion by jgillick** [community/community#43908] [Actions] [answered]
### Select Topic Area

Question

### Body

I have a [github action](https://github.com/jgillick/test-coverage-reporter) that outputs a code coverage report on the PR it's run on. In the report, I'd like it to be able to link directly to the files it's listing in the context of the PR. The URL format for this is:

```
https://github.com/<repo>/pull/<num>/files#diff-<diff>
```

For the life of me, I cannot figure out how to get the `<diff>` value of that URL. This is a 64-character string that does not seem to be a sha and isn't returned by the `pulls.listFiles` API. Is this a file ID?

I know that I could link to the standalone URL of the file; however, this would take the user out of the PR and would loose any annotations that were applied.

**Comment by jgillick** [comment, marked as answer]
I figured it out and it was just staring me in the face. The `#diff-<ID>` portion of the URL is just the file path hashed using sha256. 

For example, for the file `src/index.js`, the URL fragment would be: `#diff-bfe9874d239014961b1ae4e89875a6155667db834a410aaaa2ebe3cf89820556`

**Comment by miladrahimzadeh** [comment]
I'm not a Developer actually I'm just a IT Man which recently begin to work with Linux terminal.
I have a big question about registering an app because it ask me for url of my app I have no idea about that how I can find my url

**Reply by jgillick** [reply]
I think you're asking your question in the wrong place. I'd also recommend when you repost this to a more accurate forum, to include what you're trying to do and the steps to reproduce. As it is, your comment has nearly no helpful information for someone to answer. For example, what kind of app are you trying to register? Where is this URL field you're referring to? Is it on a web form? If so, what's the URL to this web form?
